src/proto/grpc/testing/messages.proto

```diff
@@ -212,3 +212,41 @@ message LoadBalancerStatsResponse {
   int32 num_failures = 2;
   map<string, RpcsByPeer> rpcs_by_method = 3;
 }
+
+// Request for retrieving a test client's real time stats.
+message LoadBalancerRealTimeStatsRequest {}
+
+// Real-time stats for RPCs sent by a test client.
+message LoadBalancerRealTimeStatsResponse {
+  // The real-time total number of RPCs issued.
+  int32 num_rpcs_started = 1;
+  // The real-time total number of RPCs completed successfully for each peer.
+  map<string, int32> num_rpcs_succeeded_by_peer = 2;
+  // The real-time total number of RPCs failed.
+  int32 num_rpcs_failed = 3;
+}
+
+// Configurations for a test client.
+message ClientConfigureRequest {
+
+  // Type of RPCs to send.
+  enum RpcType {
+    EMPTY_CALL = 0;
+    UNARY_CALL = 1;
+  }
+
+  // Metadata to be attached for the given type of RPCs.
+  message Metadata {
+    RpcType type = 1;
+    string key = 2;
+    string value = 3;
+  }
+
+  // The types of RPCs the client sends.
+  repeated RpcType types = 1;
```

Comment:
How do these two fields work?

RPC with `types[0]` will send `metadata[0]`?
Make another message for `{type, metadata}`?

Reply:
The field `types` and `metadata` in `ClientConfigureRequest` are orthogonal, equivalent to what we have for command line options today. Basically, `types` specifies the kinds of RPCs the test client will send and `metadata` contains a list of metadata to be attached to the kind of RPCs as specified inside it. For example, if we have `types = [UNARY, EMPTY]` and `metadata = [ {UNARY, 'key1', 'value1'}, {UNARY, 'key2', 'value2'} ]`, then the test client will send two kinds of RPCs and only the unary RPCs will have two metadata pairs.

Reply:
Oh, right. I missed that `message Metadata` also have `type` in it.